### PR TITLE
Fixed issue #18633: Inherited value doesn't match any attribute option

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/views/themeOptions/options_core.php [hunk 1]
```diff
@@ -126,7 +126,7 @@
                                         $sParentLabelOption = $options[$sParentOption];
                                         $options['inherit'] = gT("Inherit") . ' [' . gT($sParentLabelOption) . ']';
                                     } else {
-                                        $sParentOption = $sParentOption !== '' ? gT($sParentOption) : $sParentOption;
+                                        $sParentOption = isset($options[$sParentOption]) ? gT($options[$sParentOption]) : $sParentOption;
```
Comment (Shnoulle) [inline application/views/themeOptions/options_core.php:129]: usage of empty maybe ? here you seems you can go to gT("") ?

And previous system seems to send '' as value

**Review by Shnoulle — COMMENTED** [review]
Usage of empty ? "" null and 0 must not be translated

**Review by Shnoulle — APPROVED** [review]
:+1:

File: application/views/themeOptions/options_core.php [hunk 2]
```diff
@@ -126,7 +126,7 @@
                                         $sParentLabelOption = $options[$sParentOption];
                                         $options['inherit'] = gT("Inherit") . ' [' . gT($sParentLabelOption) . ']';
                                     } else {
-                                        $sParentOption = $sParentOption !== '' ? gT($sParentOption) : $sParentOption;
+                                        $sParentOption = ! empty($options[$sParentOption]) ? gT($options[$sParentOption]) : '';
```
Comment (olleharstedt) [inline application/views/themeOptions/options_core.php:129]: Can we use `??` instead please?

**Review by olleharstedt — CHANGES_REQUESTED** [review]

**Comment by olleharstedt** [comment]
Pretty big conflict, probably due to the LS6 release. Might be easier to redo the PR?

File: application/views/themeOptions/options_core.php (hunk 2 above)
Comment (Shnoulle) [reply application/views/themeOptions/options_core.php:129]: ?? is only for null, here it's for `'' `

Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:129]: No, the check is on `empty`

Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:129]: `$options[$sParentOption] ?? ''`

Comment (Shnoulle) [reply application/views/themeOptions/options_core.php:129]: > `$options[$sParentOption] ?? ''` should mean the same.

No , if `$options[$sParentOption] === ''` then you return `gT('')` we already have it in 5.X for help when help was not set by plugin (then are '' (and not null).

Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:129]: Oh. So

```php
gT($options[$sParentOption] ?? '')
```

?

Comment (olleharstedt) [reply application/views/themeOptions/options_core.php:129]: Ah not exactly the same I guess.

Comment (gabrieljenik) [reply application/views/themeOptions/options_core.php:129]: > No, the check is on `empty`

Empty returns false if string is empty.
That's why it is not the same.

**Pull request merged**